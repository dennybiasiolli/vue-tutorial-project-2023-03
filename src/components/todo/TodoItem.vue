<script setup>
const props = defineProps({
  text: {
    type: String,
    required: true,
  },
  completed: {
    type: Boolean,
    required: true,
  },
})
</script>

<template>
  <input type="checkbox" v-model="completed" /> {{ text }}
</template>
