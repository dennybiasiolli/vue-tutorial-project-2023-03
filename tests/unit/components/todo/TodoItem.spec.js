import { shallowMount } from '@vue/test-utils'
import { expect, it } from 'vitest'
import TodoItem from '@/components/todo/TodoItem.vue'

it('should render as expected', () => {
  const wrapper = shallowMount(TodoItem, {
    props: {
      text: 'farina',
      completed: false,
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.setProps({
    completed: true,
  })
  wrapper.vm.$nextTick()
  expect(wrapper.html()).toMatchSnapshot()
})

it('should emit an event', () => {
  const wrapper = shallowMount(TodoItem, {
    props: {
      text: 'farina',
      completed: false,
    },
  })
  wrapper.find('input').trigger('change')
  expect(wrapper.emitted('todoChange'))
})
