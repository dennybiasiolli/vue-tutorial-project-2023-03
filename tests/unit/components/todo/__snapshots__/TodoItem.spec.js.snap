// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render as expected 1`] = `"<span><input type=\\"checkbox\\"> farina <button>Edit</button><button>Delete</button></span>"`;

exports[`should render as expected 2`] = `"<span><input type=\\"checkbox\\"> farina <button>Edit</button><button>Delete</button></span>"`;
